client/components/section-nav/tabs.jsx

```diff
@@ -152,6 +157,16 @@ class NavTabs extends Component {
 		}
 	};
 
+	setDropdownDebounced = debounce( this.setDropdown, 300 );
+
+	// setDropdown reads element sizes from DOM. If called synchronously in the middle of a React
+	// update, it causes a synchronous layout reflow, doing the layout two or more times instead
+	// of just once after all the DOM writes are finished. Prevent that by scheduling a callback
+	// just *after* the next layout flush. Inspired by the Firefox performance best practices at
+	// https://developer.mozilla.org/en-US/Firefox/Performance_best_practices_for_Firefox_fe_engineers
+	setDropdownAfterLayoutFlush = () =>
+		requestAnimationFrame( () => setTimeout( () => this.setDropdown(), 0 ) );
```

Comment:
A setTimeout inside of a RAF is relatively uncommon to see. We might why to add an extra sentence to explain the workaround. I believe Moz has the following in one of their examples.

```
    // This code will be run ASAP after Style and Layout information have
    // been calculated and the paint has occurred. Unless something else
    // has dirtied the DOM very early, querying for style and layout information
    // here should be cheap.
```

Reply:
I added some explanations to the `afterLayoutFlush` documentation.